components/eventing-controller/pkg/handlers/utils.go

```diff
@@ -43,6 +44,49 @@ type MessagingBackend interface {
 
 const charset = "abcdefghijklmnopqrstuvwxyz0123456789"
 
+// NameMapper is used to map Kyma-specific resource names to their corresponding name on other
+// (external) systems, e.g. on different eventing backends, the same Kyma subscription name
+// could map to a different name.
+type NameMapper interface {
+	MapSubscriptionName(sub *eventingv1alpha1.Subscription) string
+}
+
+// bebSubscriptionNameMapper maps a Kyma subscription to an ID that can be used on the BEB backend,
+// which has a max length. Domain name is used to make the names on BEB unique.
+type bebSubscriptionNameMapper struct {
+	domainName string
+	maxLength  int
+}
+
+func NewBebSubscriptionNameMapper(domainName string, maxNameLength int) *bebSubscriptionNameMapper {
+	return &bebSubscriptionNameMapper{
+		domainName: strings.TrimSpace(domainName),
+		maxLength:  maxNameLength,
+	}
+}
+
+func (m *bebSubscriptionNameMapper) MapSubscriptionName(sub *eventingv1alpha1.Subscription) string {
+	hash := hashSubscriptionFullName(m.domainName, sub.Namespace, sub.Name)
+	return shortenNameAndAppendHash(sub.Name, hash, m.maxLength)
+}
+
+func hashSubscriptionFullName(domainName, namespace, name string) string {
+	hash := sha1.Sum([]byte(domainName + namespace + name))
+	return fmt.Sprintf("%x", hash)
+}
+
+func shortenNameAndAppendHash(name string, hash string, maxLength int) string {
+	maxNameLen := maxLength - len(hash)
```

Comment:
Is there a possibility that `len(hash)` might be greater than `maxLength`?

Reply:
not if the 50 is not configurable. The hash will always be 40 anyway. We could check and panic.

Reply:
In other words, we are assuming the max length is enough at least for the hash value.